### ISSUE TITLE
`mvn clean verify` doesn't work without running install.

### ISSUE THREAD

**Issue by SamBarker** [kroxylicious/kroxylicious#106]
When I ran `mvn clean verify` it wouldn't work as it told me I didn't have a specific (kroxylicious) related component installed. I'm afraid I forgot to note exactly what one it was, though (I tried this late last week) - but switching this to install worked for me. @grdryn do you happen to remember what this was?

_Originally posted by @racheljpg in https://github.com/kroxylicious/kroxylicious/pull/91#discussion_r1033382150_

There is something suspicious about needing an install, it implies the module dependencies aren't right. We should look at fixing that.

**Comment by gunnarmorling** [comment]
I suppose Maven doesn't pick up the plug-in dependency correctly. See [this SO question](https://stackoverflow.com/questions/17284553/maven-plugin-as-dependency-in-multi-module-project) for a (hacky) work-around.

**Comment by SamBarker** [comment]
Hmmm the plug-in dependency sounds like the sort of thing I thought was going on but I can't reproduce locally.

```bash
 ~kroxylicious │ main  ls -l ~/.m2/repository/io/kroxy*                                                                                                    ✔ │ system  │ 13:58:43 
zsh: no matches found: /Users/sbarker/.m2/repository/io/kroxy*
 ~kroxylicious │ main  mvn clean verify                                                                                                                  1 ✘ │ system  │ 13:59:00 
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for io.kroxylicious:integrationtests:jar:1.0-SNAPSHOT
[WARNING] The expression ${build.directory} is deprecated. Please use ${project.build.directory} instead.
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Kroxylicious Parent                                                [pom]
[INFO] kroxylicious-krpc-plugin                                  [maven-plugin]
[INFO] Kroxylicious Proxy                                                 [jar]
[INFO] integrationtests                                                   [jar]
[INFO] 
... SNIP ...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kroxylicious Parent 1.0-SNAPSHOT:
[INFO] 
[INFO] Kroxylicious Parent ................................ SUCCESS [  1.482 s]
[INFO] kroxylicious-krpc-plugin ........................... SUCCESS [  9.133 s]
[INFO] Kroxylicious Proxy ................................. SUCCESS [ 13.601 s]
[INFO] integrationtests ................................... SUCCESS [ 54.153 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:18 min
[INFO] Finished at: 2023-01-16T14:00:26+13:00
```

Which based on the [cited link](https://www.cloudbees.com/blog/maven-and-install-hack) from the SO question shouldn't work. More digging required...

**Comment by SamBarker** [comment]
@racheljpg what version of maven are you using? 

Run `mvn -v` if you need to check.

**Comment by racheljpg** [comment]
@SamBarker I'm using `Apache Maven 3.8.4 (Red Hat 3.8.4-3)`.

**Comment by SamBarker** [comment]
hrm its not something that's been fixed in a newer maven version then 😞

**Comment by SamBarker** [comment]
Taking a turn for the stranger. 

Running `mvn process-sources` fails locally but `mvn verify` passes.

```shell
[INFO] --- maven-dependency-plugin:3.3.0:unpack (unpack-message-specs) @ kroxylicious-krpc-plugin ---
[INFO] Configured Artifact: org.apache.kafka:kafka-clients:3.3.1:jar
[INFO] Unpacking /Users/sbarker/.m2/repository/org/apache/kafka/kafka-clients/3.3.1/kafka-clients-3.3.1.jar to /Users/sbarker/development/kroxy/kroxylicious/krpc-code-gen/target/message-specs with includes "common/message/*.json" and excludes ""
[INFO] 
[INFO] --------------------< io.kroxylicious:kroxylicious >--------------------
[INFO] Building Kroxylicious Proxy 1.0-SNAPSHOT                           [3/4]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Kroxylicious Parent 1.0-SNAPSHOT:
[INFO] 
[INFO] Kroxylicious Parent ................................ SUCCESS [  0.800 s]
[INFO] kroxylicious-krpc-plugin ........................... SUCCESS [  1.610 s]
[INFO] Kroxylicious Proxy ................................. FAILURE [  0.022 s]
[INFO] integrationtests ................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.565 s
[INFO] Finished at: 2023-01-18T16:08:13+13:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin io.kroxylicious:kroxylicious-krpc-plugin:1.0-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact io.kroxylicious:kroxylicious-krpc-plugin:jar:1.0-SNAPSHOT -> [Help 1]
```

**Comment by SamBarker** [comment]
fixed by #117
